app/models/devolved_administration_availability.rb

```diff
@@ -4,4 +4,27 @@ class DevolvedAdministrationAvailability
   embedded_in :local_transaction_edition
   field :type, type: String, default: "local_authority_service"
   field :alternative_url, type: String
+
+  validates :type, inclusion: { in: %w[local_authority_service devolved_administration_service unavailable], allow_blank: true }
+  validate :alternative_url_presence
+  validate :alternative_url_format
+
+  def alternative_url_presence
+    if devolved_administration_service_selected? && alternative_url.blank?
+      errors.add(:alternative_url, "must be provided")
+    end
+  end
+
+  def alternative_url_format
+    if devolved_administration_service_selected? && alternative_url.present?
```

Comment:
You might find it's a bit cleaner to handle this with a guard clause e.g. `return if alternative_url.blank? || !develoved_administration_service_selected?`

Reply:
I updated this but just the `return if alternative_url.blank?` part. Otherwise I think we could encounter the problem you mentioned above of the `alternative_url_format` not being validated if the type changes.